### PR TITLE
Map info widget

### PR DESCRIPTION
**Pull request by TRONAS**

### CODE REVIEW

**Comment by TRONAS** [comment]
screen00020

File: LuaUI/Widgets/Map_info.lua [hunk 1]
```diff
@@ -0,0 +1,272 @@
...
+local glPushMatrix      = gl.PushMatrix
+local glPopMatrix       = gl.PopMatrix
+local glTranslate       = gl.Translate
+local glBeginEnd        = gl.BeginEnd
+local glVertex          = gl.Vertex
+local glGetTextWidth	= gl.GetTextWidth
+local glBlending		= gl.Blending
+
+local glDepthTest       = gl.DepthTest
+local glAlphaTest       = gl.AlphaTest
```
Comment (sprunk) [inline LuaUI/Widgets/Map_info.lua:56]: unused

File: LuaUI/Widgets/Map_info.lua [hunk 2]
```diff
@@ -0,0 +1,272 @@
...
+local glTexture         = gl.Texture
```
Comment (sprunk) [inline LuaUI/Widgets/Map_info.lua:57]: unused

File: LuaUI/Widgets/Map_info.lua [hunk 3]
```diff
@@ -0,0 +1,272 @@
...
+local glRotate          = gl.Rotate
+
+local mapInfo = {}
+local mapInfoWidth = 400	-- minimum width
+
+--------------------------------------------------------------------------------
+-- Functions
+--------------------------------------------------------------------------------
+
+function DrawMapInfo(backgroundOpacity, opacityMultiplier)
```
Comment (sprunk) [inline LuaUI/Widgets/Map_info.lua:67]: local

Comment (sprunk) [inline LuaUI/Widgets/Map_info.lua:67]: `backgroundOpacity` shadows the one from line 27

File: LuaUI/Widgets/Map_info.lua [hunk 4]
```diff
@@ -0,0 +1,272 @@
...
+	
+	glDepthTest(true)
+	glBlending(GL.SRC_ALPHA, GL.ONE_MINUS_SRC_ALPHA)
+	
+	if stickToFloor then
+		glTranslate(offset, offset, Game.mapSizeZ)
+	else
+		glTranslate(offset, mapInfoBoxHeight-offset, Game.mapSizeZ)
```
Comment (sprunk) [inline LuaUI/Widgets/Map_info.lua:75]: `mapInfoBoxHeight` is needlessly a global

File: LuaUI/Widgets/Map_info.lua [hunk 5]
```diff
@@ -0,0 +1,272 @@
...
+	glScale(scale,scale,scale)
+	if backgroundOpacity > 0 then
+		glRotate(90,0,-1,0)
+		glRotate(180,1,0,0)
+		
+		local length = math.max(mapInfoWidth, (glGetTextWidth(mapInfo.mapDescription)*12) + 45)
+		--Spring.Echo(glGetTextWidth(mapInfo.mapDescription))
+		
+		local height = 90
+		local thickness = -(thickness*scale)
```
Comment (sprunk) [inline LuaUI/Widgets/Map_info.lua:87]: thickness shadows the earlier thickness

File: LuaUI/Widgets/Map_info.lua [hunk 6]
```diff
@@ -0,0 +1,272 @@
...
+		glRotate(180,1,0,0)
+		glRotate(90,0,1,0)
+	end
+	
+	glRotate(90,1,0,0)
+	glTranslate(0,3,0)
+	
+	local textOffsetX = 11
+	local textOffsetY = 16
+	usedTextOffsetY = textOffsetY + (offset/2)
```
Comment (sprunk) [inline LuaUI/Widgets/Map_info.lua:132]: local

File: LuaUI/Widgets/Map_info.lua [hunk 7]
```diff
@@ -0,0 +1,272 @@
...
+		glTranslate(-offset, -offset, -Game.mapSizeZ)
+	else
+		glTranslate(-offset, -mapInfoBoxHeight-offset, -Game.mapSizeZ)
+	end
+	glColor(1,1,1,1)
+	glScale(1,1,1)
+	glDepthTest(false)
+end
+
+function Init()
```
Comment (sprunk) [inline LuaUI/Widgets/Map_info.lua:194]: local

File: LuaUI/Widgets/Map_info.lua [hunk 8]
```diff
@@ -0,0 +1,272 @@
...
+		if opacityMultiplier > 0.06 then
+			DrawMapInfo(backgroundOpacity, opacityMultiplier)
+		end
+	end
+end
+
+
+
+function widget:GetConfigData(data)
+    savedTable = {}
```
Comment (sprunk) [inline LuaUI/Widgets/Map_info.lua:258]: local

File: LuaUI/Widgets/Map_info.lua [hunk 9]
```diff
@@ -0,0 +1,272 @@
...
+    savedTable.stickToFloor = stickToFloor
+    return savedTable
+end
+
+function widget:SetConfigData(data)
+    if data.stickToFloor ~= nil 	then  stickToFloor	= data.stickToFloor end
```
Comment (sprunk) [inline LuaUI/Widgets/Map_info.lua:264]: crappy whitespace, needless explicit nil check

File: LuaUI/Widgets/Map_info.lua [hunk 10]
```diff
@@ -0,0 +1,272 @@
...
+end
+
+function widget:TextCommand(command)
+    if (string.find(command, "mapinfo_floor") == 1  and  string.len(command) == 13) then
```
Comment (sprunk) [inline LuaUI/Widgets/Map_info.lua:268]: trailing whitespace

**Pull request closed without merging**